### ISSUE TITLE
How to setup With Username and Password?

### ISSUE THREAD

**Issue by Hellangel1987** [BenRoe/MMM-FHEM#2]
Hi,

how to set this up with username and password for fhem ?

Try to set it up as shown in readme. Not showing any values - but smart mirror will boot normally.

tried also:

  		{
    			module: 'MMM-FHEM',
    			position: 'bottom_bar',
    			config: {
			user: 'USERNAME',
			password: 'PASSWORD',
      			deviceNames: [
                    	'Arbeitszimmer',
                    	'Wohnzimmer',
                    	'Schlafzimmer',
			'Kueche',
			'Bad',
                ],
      			deviceReadings: [
                        { name: 'state', icon: 'wi wi-thermometer', suffix: '&deg;' },
                 ],
    		},

also no success....

**Comment by BenRoe** [comment]
Currently it's not supported. Will try to add this in the next update.

**Comment by Da-ne-ezy** [comment]
Yes an update with USERNAME and PASSWORD would be great. No fixes yet?

**Comment by Hellangel1987** [comment]
We Seriosly need a fix for Username and Password otherwise this module is useless @BenRoe  any plans ?

ALL Fhem User use Username and Passwort for their Enviroment....

**Comment by hliebe** [comment]
The setup of username and password can be done by defining your host like this:
host: 'username:password@ip-of-your-fhem-server'

**Comment by BenRoe** [comment]
Hi hliebe,
thanks for the hint.
I don't have a mm running. Can someone (@Hellangel1987 @Da-ne-ezy )please test this?
If it works i will add it to the description an close this issue.

**Comment by Hellangel1987** [comment]
I will Test this - give me a Couple of days

**Comment by Da-ne-ezy** [comment]
I tried it like this, but the module doesn't even appear on my mirror.  Maybe any changes on the allowed_WEB?

**Comment by PatiB87** [comment]
/push This is not working for me. The widget is created on the magicmirror-webpage but it stays empty.

**Comment by Aniflu** [comment]
@Da-ne-ezy 

> I tried it like this, but the module doesn't even appear on my mirror. Maybe any changes on the allowed_WEB?

the Fix for you`r Problem might be:

[Disable csrfToken](https://forum.magicmirror.builders/topic/1566/mmm-fhem-no-fhem-statusdisplay/11?page=2)

to test the module you can insert in the config  '**header: 'Smarthome',**' to see if the module is pop up an header if not the problem might be somwere else. 

as an last option you can try ` attr WEB.* csrfToken none`

**Comment by PieBa** [comment]
For me it works with host: 'username:password@ip-of-your-fhem-server' as suggested by @hliebe 
I needed to deactivate csrfToken as mentioned by @Aniflu 

However I was not able to use the module with https: true
